backbone/src/test/java/org/researchstack/backbone/ui/task/TaskViewModelTest.kt

```diff
@@ -0,0 +1,419 @@
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(nextStepMocked)
+        `when`(taskViewModel.isReviewStep(nextStepMocked)).thenReturn(true)
+
+        // Act
+        taskViewModel.removeUpdatedLayout()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.editing, false)
+        Assert.assertEquals(taskViewModel.updateCancelEditInLayout.value, true)
+        Assert.assertEquals(taskViewModel.hideMenuItemCancel.value, false)
+    }
+
+    @Test
+    fun testNext_ShouldUpdateCurrentStepToNextStep_shouldUpdateCurrentStepEventToNextStep() {
+        // Assemble
+        taskViewModel = createViewModel()
+        val currentStepMocked = createCurrentStep()
+        val nextStepMocked = createCurrentStep("nextStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepAfterStep(any(), any())).thenReturn(nextStepMocked)
+
+        // Act
+        taskViewModel.nextStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, nextStepMocked)
+        Assert.assertEquals(taskViewModel.currentStepEvent.value!!.step, nextStepMocked)
+    }
+
+    @Test
+    fun testPrevious_ShouldUpdateCurrentStepToPreviousStep_shouldUpdateCurrentStepEventToPreviousStep() {
+        // Assemble
+        taskViewModel = createViewModel()
+
+        val currentStepMocked = createCurrentStep()
+        val previousStepMocked = createCurrentStep("previousStepMocked")
+        taskViewModel.currentStep = currentStepMocked
+        `when`(task!!.getStepBeforeStep(any(), any())).thenReturn(previousStepMocked)
+
+        // Act
+        taskViewModel.previousStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.currentStep, previousStepMocked)
+        Assert.assertEquals(taskViewModel.currentStepEvent.value!!.step, previousStepMocked)
+    }
+
+    @Test
+    fun checkPrevious_shouldCloseWhenPreviousIsNull() {
+        // Assemble
+        taskViewModel = createViewModel()
+        `when`(task!!.getStepBeforeStep(any(), any())).thenReturn(null)
+        val currentStepMocked = createCurrentStep()
+        taskViewModel.currentStep = currentStepMocked
+
+        // Act
+        taskViewModel.previousStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.taskCompleted.value, false)
+    }
+
+    @Test
+    fun testNext_shouldCloseWhenNextNull() {
+        // Assemble
+        taskViewModel = spy(createViewModel())
+
+        // Act
+        taskViewModel.nextStep()
+
+        // Assert
+        Assert.assertEquals(taskViewModel.taskCompleted.value, true)
+    }
+
+    @Test
+    fun testShowCurrentStep_ifCurrentStepIsNull_ShowGoToNextStep() {
+        // Assemble
+        taskViewModel = spy(createViewModel())
+        // Act
+        taskViewModel.showCurrentStep()
+
+        // Assert
+        verify(taskViewModel, times(1)).showCurrentStep()
+    }
+
+    @Test
+    fun testPreviousOnEditMode_shouldCheckForCancelDialog() {
```

Comment:
I believe you need to rename this one too, it's not 100% clear and it doesn't match the assertions:

`whenEditingAStep_previousStep_shouldShowCancelDialog` is (in my opinion) much more clear. 

**UNLESS**: you're just testing if it was checked, in which case, you'd need to test that `checkIfAnswersAreTheSame()` was called too...